File: test-suite/src/test/java/io/micronaut/docs/inject/generics/V6Engine.java
package io.micronaut.docs.inject.generics;

import jakarta.inject.Singleton;

// tag::class[]
@Singleton
public class V6Engine implements Engine<V6> {  // <1>
    @Override
    public V6 getCylinderProvider() {
        return new V6();
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/qualifiers/annotationmember/V6Engine.py
from jakarta.inject import Singleton
from .Engine import Engine
from .Cylinders import Cylinders

# tag::class[]
@Singleton
@Cylinders(value = 6, description = "6-cylinder V6 engine")
class V6Engine(Engine):
    cylinders: int = 6

    def start(self) -> str:
        return "Starting V6"

    def get_cylinders(self) -> int:
        return self.cylinders
# end::class[]
